src/govuk/components/button/button.yaml

```diff
@@ -148,6 +148,33 @@ examples:
       text: Warning button
       href: '/'
       classes: govuk-button--warning
+  - name: Inverse
+    description: A button that appears on dark backgrounds
+    data:
+      name: Inverse
+      text: Inverse button
+      classes: govuk-button--inverse
```

Comment:
I'm open to being disagreed with as I know this is an approach we take elsewhere but my personal preference would be to control this via an API attribute eg: `inverse: true`. Communicating features in this way feels clearer to me. It would then be more justified to write a test as you're testing that the API successfully applies the class.

Reply:
As the stylistic alternatives are all mutually exclusive, I'd be inclined to combine them into a single option so that it's not possible to enable more than one (e.g. `appearance: "inverse"`), but then we'd probably have to add documentation what the allowed values are, as it may not be particularly obvious otherwise.

Reply:
That's a good point @36degrees , dropping this point for now. @querkmachine I like this idea on principal but I also think this is something we should apply strategically to our whole codebase. I'll write up an issue to explore this.

Reply:
Can I get a review if we're happy to leave as is? 🙏